inngest-spring-boot-demo/src/test/java/com/inngest/springbootdemo/SyncRequestTest.java

```diff
@@ -0,0 +1,130 @@
+import com.inngest.*;
+import com.inngest.springboot.InngestConfiguration;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.org.webcompere.systemstubs.environment.EnvironmentVariables;
+import uk.org.webcompere.systemstubs.jupiter.SystemStub;
+import uk.org.webcompere.systemstubs.jupiter.SystemStubsExtension;
+
+import java.util.HashMap;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+
+@ExtendWith(SystemStubsExtension.class)
+public class SyncRequestTest {
+    static class SyncInngestConfiguration extends InngestConfiguration {
+        protected HashMap<String, InngestFunction> functions() {
+            return new HashMap<>();
+        }
+
+        @Override
+        protected Inngest inngestClient() {
+            return new Inngest("spring_test_registration");
+        }
+
+        @Override
+        protected ServeConfig serve(Inngest client) {
+            return new ServeConfig(client);
+        }
+
+        @Bean
+        protected CommHandler commHandler(@Autowired Inngest inngestClient) {
+            ServeConfig serveConfig = new ServeConfig(inngestClient);
+            return new CommHandler(functions(), inngestClient, serveConfig, SupportedFrameworkName.SpringBoot);
+        }
+    }
+
+    @SystemStub
+    private static EnvironmentVariables environmentVariables;
+
+    public static MockWebServer mockWebServer;
+
+    @Import(SyncInngestConfiguration.class)
+    @WebMvcTest(DemoController.class)
+    @Nested
+    @EnabledIfSystemProperty(named = "test-group", matches = "unit-test")
+    @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+    class InnerSpringTest {
+        @Autowired
+        private MockMvc mockMvc;
+
+        @BeforeEach
+        void BeforeEach() throws Exception {
+            mockWebServer = new MockWebServer();
+            mockWebServer.start();
+
+            String serverUrl = mockWebServer.url("").toString();
+
+            environmentVariables.set("INNGEST_API_BASE_URL", serverUrl.substring(0, serverUrl.length() - 1));
+        }
+
+        @AfterEach
+        void afterEach() throws Exception {
+            mockWebServer.shutdown();
+        }
+
+        private void assertThatPayloadDoesNotContainDeployId(RecordedRequest recordedRequest) throws Exception {
```

Comment:
https://github.com/inngest/inngest/blob/main/docs/SDK_SPEC.md#432-syncing
>If a deployId query string parameter is set on the sync request received by the SDK, the SDK MUST set the same key and value on the query string of the POST request to Inngest and MUST NOT include the deployId in the URL included in the Sync’s payload. The deployId is used to attribute various parts of a sync’s handshake with each other.

I could add a comment to the doc but i'm not sure if it's a good idea.